### PR TITLE
fix: Use correct endpoints

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**
Read commit by commit:
- (417cc90fabb154f77d13f951ed00a9a89d76d902) Trash -> restore / delete permanently use V2
- (12387860dc3153893c21d3d95fdb98a4024f603a) Activities for file details -> use cursor
- (5391da64ea1b9e13bcd68d00e0a6447f7437a681) Remove dead code
- (f6df90dfabfca9a40956c36a068b180da95a2964) Remove default argument for searchFiles

### CODE REVIEW

File: kDriveAPITests/kDriveCore/DriveApiTests.swift [hunk 1]
```diff
@@ -597,34 +597,14 @@ final class DriveApiTests: XCTestCase {
...
-            drive: proxyDrive,
-            files: [file, secondFile],
-            from: Date(timeIntervalSince1970: 0)
-        ).validApiResponse.data
-        XCTAssertEqual(activities.count, 2, "Array should contain two activities")
-        for activity in activities {
-            XCTAssertTrue(activity.result, TestsMessages.shouldReturnTrue)
-            XCTAssertNil(activity.message, TestsMessages.noError)
-        }
+        _ = try await currentApiFetcher.fileActivities(file: file, from: earlyDate, cursor: nil)
```
Comment (adrien-coye) [inline kDriveAPITests/kDriveCore/DriveApiTests.swift:607]: No assertions on the result of this call?

**Review by adrien-coye — COMMENTED** [review]
Thanks for fixing some routes, I only got one tiny comment.

File: kDriveAPITests/kDriveCore/DriveApiTests.swift (hunk 1 above)
Comment (PhilippeWeidmann) [reply kDriveAPITests/kDriveCore/DriveApiTests.swift:607]: I think the goal of this test was juste to check if we could do the API call and decode the result correctly

**Review by adrien-coye — APPROVED** [review]
All set.

**Pull request merged**